fearless/Common/Storage/StakingAssetSettings.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+import SoraKeystore
+import RobinHood
+
+final class StakingAssetSettings: PersistentValueSettings<ChainAsset> {
+    let settings: SettingsManagerProtocol
+    let operationQueue: OperationQueue
+
+    init(
+        storageFacade: StorageFacadeProtocol,
+        settings: SettingsManagerProtocol,
+        operationQueue: OperationQueue
+    ) {
+        self.settings = settings
+        self.operationQueue = operationQueue
+
+        super.init(storageFacade: storageFacade)
+    }
+
+    override func performSetup(completionClosure: @escaping (Result<ChainAsset?, Error>) -> Void) {
+        let repository: AnyDataProviderRepository<ChainModel>
+        let mapper = AnyCoreDataMapper(ChainModelMapper())
+
+        let maybeChainAssetId = settings.stakingAsset
+
+        if let chainAssetId = maybeChainAssetId {
+            let filter = NSCompoundPredicate(orPredicateWithSubpredicates: [
+                NSPredicate.chainBy(identifier: chainAssetId.chainId),
+                NSPredicate.relayChains()
+            ])
+
+            repository = AnyDataProviderRepository(
+                storageFacade.createRepository(filter: filter, sortDescriptors: [], mapper: mapper)
+            )
+        } else {
+            let filter = NSPredicate.relayChains()
+            repository = AnyDataProviderRepository(
+                storageFacade.createRepository(filter: filter, sortDescriptors: [], mapper: mapper)
+            )
+        }
+
+        let fetchOperation = repository.fetchAllOperation(with: RepositoryFetchOptions())
+
+        let mappingOperation = ClosureOperation<ChainAsset?> {
+            let chains = try fetchOperation.extractNoCancellableResultData()
+
+            if
+                let selectedChain = chains.first(where: { $0.chainId == maybeChainAssetId?.chainId }),
+                let selectedAsset = selectedChain.assets.first(where: { $0.assetId == maybeChainAssetId?.assetId }) {
+                return ChainAsset(chain: selectedChain, asset: selectedAsset)
+            }
+
+            let relayChains = chains.filter { $0.assets.contains { $0.isUtility }}
+
+            if
+                let firstRelayChain = relayChains.min(by: { $0.addressPrefix < $1.addressPrefix }),
+                let asset = firstRelayChain.assets.first(where: { $0.isUtility }) {
+                self.settings.stakingAsset = ChainAssetId(
+                    chainId: firstRelayChain.chainId,
+                    assetId: asset.assetId
+                )
+                return ChainAsset(chain: firstRelayChain, asset: asset)
+            }
+
+            return nil
```

Comment:
Looks like this `if` block could be replaced with 
```swift
guard let ...
else { return nil }

self.settings.stakingAsset = ChainAssetId(
    chainId: firstRelayChain.chainId,
    assetId: asset.assetId
)
return ChainAsset(chain: firstRelayChain, asset: asset)
```

Reply:
This block of code consists from sequence of `if` statements. I think if we use guard instead of second `if` then it will become less readable